qmf/src/libraries/qmfclient/qmailmessage.cpp

```diff
@@ -4169,7 +4179,11 @@ QByteArray QMailMessagePartContainerPrivate::boundary() const
 
 void QMailMessagePartContainerPrivate::setBoundary(const QByteArray& text)
 {
-    _boundary = text;
+    if (text == QByteArray("NIL")) {
```

Comment:
We will just not try to fetch it and crash in the above code, so in this case we will just fetch a body peak(BODY.PEEK[]) get the structure from the RFC822.HEADERS and happily proceed from there with correct parts.